bounding box and variable selection

maximusjstevens opened the pull request:
retrieve_data.sh: I changed the wget command. Now includes option to specify directory where files are saved. It also now calls using different options, and the filenames get truncated to not include the variable names and bounding box indices, etc.

new file: M2LL.nc (I overrode the .gitignore for this file). This is just the full MERRA-2 lat/lon grid. It is used to figure out the indices that correspond with max/min lat and lon as specified so that a bounding box can be specified in the wget call.

print_wget_commands.py: (1) script now uses the bounding box provided in download_MERRA2.sh and finds the indices in the MERRA-2 grid corresponding to that bounding box. (2) Variables to be downloaded for each dataset (e.g. surface) are specified in a list. The slicing to get temporal/spatial extent using indices is automated for each variable, and those are concatenated. (3) Switched to use of string interpolation/f-strings to improve readability.

maximusjstevens committed:
bounding box and variable selection

maximusjstevens committed:
small changes

EricKeenan commented on download_MERRA2.sh:
```diff
@@ -41,6 +41,6 @@ python3 print_wget_commands.py ${year_start} ${year_end} ${month_start} ${month_
-mkdir -p nc_files
-rm nc_files/*.nc
-csh create_forcing_nc.sh $year_start $year_end
+# mkdir -p nc_files
+# rm nc_files/*.nc
+# csh create_forcing_nc.sh $year_start $year_end
```
Do these changes have an intended purpose?

maximusjstevens replied:
I commented those out for my own use; I didn't need the post processing of the downloaded files. The comment in it is a bit unclear - wget is retrieving netCDF files; is this concatenating all of the several thousand daily files into yearly files? If possible it would be good in this script to change the hard coded paths to variables set at the beginning.

EricKeenan commented:
@maximusjstevens: I think your updated `*/wget_*.sh` lists (e.g. `radiation/wget_radiation.sh`) are larger than they need to be to serve as an example. What do you think?

EricKeenan commented on retrieve_data.sh:
```diff
@@ -1,22 +1,30 @@
 
+# Option to specify where to save the MERRA-2 files (MS)
+dpth='/Volumes/Samsung_T1/MTEST' #save in present directory
+# dpth='/PATH/TO/DIRECTORY' #e.g. if you want to put on an externa drive
```
Is this a required setting? If so, consistent with the instructions in `README.md`, this should be defined in `download_MERRA2.sh`

maximusjstevens replied:
This is specifying the path of where to download the data to - I wanted it going straight to an external hard drive. Agree it should be specified in download_MERRA2.sh; I was too lazy with my mediocre shell scripting skills to code it to pass a variable.

EricKeenan commented on retrieve_data.sh:
```diff
@@ -1,22 +1,30 @@
-rm MERRA2*
-rm tmp*
-wget -c -nc --load-cookies ~/.urs_cookies --save-cookies ~/.urs_cookies --auth-no-challenge=on --keep-session-cookies -i wget_evap.sh
+# cd evap
+# rm MERRA2*
+# rm tmp*
+# wget -P $dpth --load-cookies ~/.urs_cookies --save-cookies ~/.urs_cookies --auth-no-challenge=on --keep-session-cookies --content-disposition -i wget_evap.sh
+# cd ..
```
Here and elsewhere: Looks like this commit will remove evap, etc downloads. Can you add them back? Or am I missing something?

maximusjstevens replied:
Yes, again this was for my own use - I didn't want to download everything in one go, e.g. I just wanted to get the radiation fields.

EricKeenan commented:
@maximusjstevens Thanks for adding your features! I know a few people in our group have used this repo for different spatial bounding boxes. So adding this feature is very useful! Thanks! 

A few of your changes seemed too specific to be merged into `master`. Can you update those? Or explain why they are necessary? Looking forward to merging!

maximusjstevens commented:
> @EricKeenan sure thing! I couldn't figure out how to comment above - but agree that the wget lists don't need to be 15000 lines :) 

I can work on updating the changes to be less specific for my use.

maximusjstevens committed:
create_forcing_nc.sh:
 - added datapath as variable, allows files to be anywhere and not hardcoded
 - added rm at end to get rid of tmp files
 - commented out prefix lines for general use
 - changed cdo calls to exclude hard-coded path to cdo

download_MERRA2.sh:
 - added datapath as variable that gets passed to subscripts (retrieve_data.sh, create_forcing_nc.sh)

retrieve_data.sh:
 - datapath is now passed in as a variable instead of being coded in this script.

general:
 - edited time-span variables to that wget_*.sh scripts are only a few lines long for example sake.
